Automattic/jetpack
# Connection: disconnect connection owner upon removal

When connected users are removed API requests are sent to WPCOM to unlink them.
It's already hooked into `deleted_user` and `remove_user_from_blog`, and works fine.

However, there are two limitations:
1. It doesn't happen for connection owners, they are skipped. Secondary users are disconnected just fine.
2. The hooks are initialized in Jetpack plugin, so standalone plugins are ignored.

## Proposed changes:
* Make sure connection owner is disconnected on WPCOM if their WordPress user account is removed.
* Move the hooks from Jetpack plugin into Connection package to also cover standalone plugins.

### Other information:


## Jetpack product discussion
p1712570085579179-slack-CDLH4C1UZ

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack under one WordPress user account (user A).
2. Create another account (user B), logout and login under that user.
3. Go to "Users" and delete the user A (connection owner). Ignore the "Connect to WordPress.com" banner.
4. Run Jetpack Debugger over the site. Confirm that it says `Everything looks great! (site-only connection)`
